src/Control/Effect/Interpret.hs

```diff
@@ -23,27 +77,27 @@ import Control.Monad.Trans.Class
 --   At time of writing, a simple passthrough use of 'runInterpret' to handle a 'State' effect is about five times slower than using 'StateC' directly.
 --
 --   prop> run (runInterpret (\ op -> case op of { Get k -> k a ; Put _ k -> k }) get) === a
-runInterpret :: (forall x . eff m x -> m x) -> InterpretC eff m a -> m a
-runInterpret handler = runReader (Handler handler) . runInterpretC
-
-newtype InterpretC eff m a = InterpretC { runInterpretC :: ReaderC (Handler eff m) m a }
-  deriving (Alternative, Applicative, Functor, Monad, MonadFail, MonadFix, MonadIO, MonadPlus)
+runInterpret
+  :: forall eff m a.
+     ( HFunctor eff, Monad m )
+  => ( forall x. eff m x -> m x )
+  -> ( forall s. Reifies s ( Handler eff m ) => InterpretC s eff m a )
+  -> m a
```

Comment:
We don’t use this specific style of signature formatting (much? at all?) in `fused-effects`, but I’ve been convinced that it’s preferable to our existing style, so good stuff 👍